coalib/bearlib/abstractions/Linter.py

```diff
@@ -677,6 +682,8 @@ def __repr__(self):
             return '<{} linter object (wrapping {!r}) at {}>'.format(
                 type(self).__name__, self.get_executable(), hex(id(self)))
 
+    LinterParent.register(LinterBase)
```

Comment:
actually doing this would register `LinterBase` again and again.

We should register `result_klass` before returning imo. Or we could just do
```
class LinterBase(LinterClass, metaclass=LinterMeta):
```
though not sure what's better^^